### ISSUE TITLE
Do not import base namespaces

### ISSUE THREAD

**Issue by iurisilvio** [soapteam/soapfish#94]
Some schemas import XSD namespaces, like `http://schemas.xmlsoap.org/soap/envelope/` and `http://schemas.xmlsoap.org/soap/encoding/`.

What is the right way to avoid import these URLs? Per domain? Any URL in `namespaces` module? Only these two hard-coded?

```xml
<xs:import
    namespace="http://schemas.xmlsoap.org/soap/envelope/"
    schemaLocation="http://schemas.xmlsoap.org/soap/envelope/" />
<xs:import
    namespace="http://schemas.xmlsoap.org/soap/encoding/"
    schemaLocation="http://schemas.xmlsoap.org/soap/encoding/" />
```

As a side note, `xsd2py` generate broken code with them.

**Comment by pope1ni** [comment]
@oskrkal: might this be addressed by your [xsd-resolver](/oskrkal/soapfish/tree/xsd-resolver) branch?

**Comment by oskrkal** [comment]
@pope1ni:  I'm sorry for late response, I somehow missed your comment.

I'm not sure if I understand the problem correctly. Is it that the schemaLocation attribute is not mandatory and only the namespace is sometimes specified in the import attribute? If so, the XSDCachedSchemaResolver indeed could address this issue. It allows to pre-define schema location for specific namespaces and is then able to resolve the schema purely by the namespace, if location is not specified.

Another feature of the resolver is that it caches all schemas, so it can save some time when, for example, a WSDL file imports a lot of schemas, most of which import a common set of other schemas.

The branch should be ready for merging into upstream. I just thought I would go through it and fix some of the constructions that you fixed in my last pull request, but I have been quite busy at work recently.

**Comment by pope1ni** [comment]
@iurisilvio: can you clarify - my understanding from what you have written was that these shouldn't be imported at all?

**Comment by iurisilvio** [comment]
Yes, I think `soap/envelope` and `soap/encoding` should be ignored. We implement them internally.

I have some third-party WSDLs with these imports, I can't use soapfish with them. Even if the generated code is not broken, it is still a lot of useless code.

These namespaces are in https://github.com/soapteam/soapfish/blob/master/soapfish/namespaces.py#L22, maybe we can use it to define these "ignored schemas".

**Comment by oskrkal** [comment]
All right, thanks for clarification. Then I think `XSDSchemaResolver`/`XSDCachedSchemaResolver` in my xsd-resolver branch is not the right place to address this problem. Its purpose is simply to load a schema file from correct location, parse it and return it to the caller, or to return a cached `xsdspec.Schema` instance if the schema has already been processed.

We probably need to somehow preload internally implemented schemas into `known_paths` list and ensure that [`get_type`](https://github.com/soapteam/soapfish/blob/master/soapfish/utils.py#L123) function returns correct module and class name.
